cmd/agent/api/internal/config/endpoint.go

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package config defines the config endpoint of the IPC API Server.
+package config
+
+import (
+	"encoding/json"
+	"expvar"
+	"fmt"
+	"net/http"
+
+	gorilla "github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type authorizedSet map[string]struct{}
+
+var authorizedConfigPaths = authorizedSet{
+	"api_key": {},
+}
+
+// runtime metrics about the config endpoint usage
+var (
+	successExpvar      *expvar.Map
+	unauthorizedExpvar *expvar.Map
+	unsetExpvar        *expvar.Map
+	failedExpvar       *expvar.Map
+	expvars            *expvar.Map
+)
+
+func init() {
+	expvars = expvar.NewMap("config_endpoint")
+	for name, expv := range map[string]**expvar.Map{
+		"success":      &successExpvar,
+		"unauthorized": &unauthorizedExpvar,
+		"unset":        &unsetExpvar,
+		"failed":       &failedExpvar,
+	} {
+		(*expv) = expvar.NewMap(name)
+		expvars.Set(name, *expv)
+	}
+}
```

Comment:
> It is a shame we are trying to prevent the use of the init function, and we are the ones adding them as well 😢

Definitely agree...
I considered it would be moved into the upcoming API component, which will remove global variables and init functions, so this way was easier 😅